### ISSUE TITLE
EnergyManagementSystem:Sensor cannot have empty key anymore in 8.9.0

### ISSUE THREAD

**Issue by jmarrec** [NREL/EnergyPlus#6520]
Issue overview
--------------

Found via the OpenStudio-resources test 'ems.rb'. (see commit [here](https://github.com/jmarrec/OpenStudio-resources/commit/85d09e3ac36673a14b04bedb662e9539365b9c84))

Here is the defect file in question: [ems.rb_890.idf.txt](https://github.com/NREL/EnergyPlus/files/1802734/ems.rb_890.idf.txt)

**Line 4842**

```
EnergyManagementSystem:Sensor,
  OATdb_Sensor,                           !- Name
  ,                                       !- Output:Variable or Output:Meter Index Key Name
  Site Outdoor Air Drybulb Temperature;   !- Output:Variable or Output:Meter Name
```

This used to run fine on 8.8.0. With current develop (537dda002df3d9f23cbbf6e34fc90359dcb925c3), it Fatals out:

```
   ** Severe  ** Invalid Output:Variable or Output:Meter Index Key Name =
   **   ~~~   ** For Output:Variable or Output:Meter = SITE OUTDOOR AIR DRYBULB TEMPERATURE
   **   ~~~   ** Entered in EnergyManagementSystem:Sensor=OATDB_SENSOR
   **   ~~~   ** Unique Key Name not found.
   **  Fatal  ** Errors found in processing Energy Management System input. Preceding condition causes termination.
   ...Summary of Errors that led to program termination:
```

I have to explicitly enter `*` as follows:

```
EnergyManagementSystem:Sensor,
  OATdb_Sensor,                           !- Name
  *,                                      !- Output:Variable or Output:Meter Index Key Name
  Site Outdoor Air Drybulb Temperature;   !- Output:Variable or Output:Meter Name
```

### Details
Some additional details for this issue (if relevant):
 - Platform: Ubuntu 16.04
 - Version of EnergyPlus: current develop 537dda002df3d9f23cbbf6e34fc90359dcb925c3

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by shorowit** [comment]
I've seen this as well. Hope it gets fixed in time for the release, unless it's intentional and meant to be covered by version transition. @Myoldmopar @mbadams5

**Comment by jmarrec** [comment]
I might add that I have checked it, and the Transition Utility of 8.9.0 does replace empty with `*`. We could take care of that in OpenStudio Forward Translator easily, but as a long time user of E+, I still find this change confusing...

**Comment by brianlball** [comment]
This is also true for output:Variable
it used to work without the * keyValue, now "Site Exterior Beam Normal Illuminance", etc are requiring it.

@jmarrec @shorowit @mjwitte @Myoldmopar

**Comment by brianlball** [comment]
specifically, 
8.8 this works:
Output:Variable,
  ,                                       !- Key Value
  Site Exterior Horizontal Sky Illuminance, !- Variable Name
  Hourly;                                 !- Reporting Frequency
8.9 it has to be
Output:Variable,
  *,                                       !- Key Value
  Site Exterior Horizontal Sky Illuminance, !- Variable Name
  Hourly;                                 !- Reporting Frequency

There is also no warning or error or any info that your outputVariable is not going thru in 8.9 if you dont have an *

**Comment by mjwitte** [comment]
@mbadams5

**Comment by mjwitte** [comment]
Actually, not sure @mbadams5 had anything to do with this. It's standard for transition to add * when it finds blank keyname. So, not sure yet why the behavior has changed. The default in the IDD for Output:Variable, Key Value is "*" so blank **should** work the same as *. Hmmm.

**Comment by brianlball** [comment]
yeah, it will execute (no error or warning) in 8.9 with blank outputVariable, but the data doesnt get put in the sql file.
[idfs.zip](https://github.com/NREL/EnergyPlus/files/1808698/idfs.zip)
here are 2 models if it helps, one with * and the other blank for some outputvars

**Comment by mbadams5** [comment]
@jmarrec @brianlball @shorowit Is this fixed due to changes in #6548? When I tested with current develop, @jmarrec original file works properly now.

**Comment by jmarrec** [comment]
It does work with the official 8.9.0 now.
The “bug” appears only on OS 2.4.3 which was based on an interim 8.9.0 build

——
Sent from a mobile device, please excuse the brevity.
Julien Marrec, EBCP, BPI MFBA
Owner


Direct: +33 6 95 14 42 13
Website: www.effibem.com

LinkedIn (en) | (fr)
